core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/PercentilesMethod.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.metrics.percentiles;
+
+import java.util.Locale;
+
+public enum PercentilesMethod {
+    TDIGEST("tdigest"), HDR("hdr");
+
+    private String name;
+
+    private PercentilesMethod(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public static PercentilesMethod resolveFromName(String name) {
+        String lowercaseName = name.toLowerCase(Locale.getDefault());
```

Comment:
We should not use the default locale to have consistent behaviours on different machines. Can we use Locale.ROOT instead?